colmena/task_server/base.py

```diff
@@ -124,8 +124,18 @@ def perform_callback(self, future: Future, result: Result, topic: str):
             topic: Topic used to send back to the user
         """
 
+        # Get any exception thrown by the workflow engine
         task_exc = future.exception()
 
+        # If it was, send back a modified copy of the input structure
+        if task_exc is not None:
+            # Mark it as unsuccessful and capture the exception information
+            result.success = False
+            result.failure_info = FailureInformation.from_exception(task_exc)
+        else:
+            # If not, the result object is the one we need
+            result = future.result()
+
         # The task could have failed at the workflow engine level (task_exc)
         # or application level (result.failure_info)
         task_failed = (task_exc is not None) or (result.failure_info is not None)
```

Comment:
We don't need to check `task_exc` here anymore right? Just `result.failure_info` as we've inserted `task_exc` into `result.failure_info` is `task_exc` is not `None`.

Reply:
Nice catch. I updated it in the latest commit, I think all we need to check is `not result.success`. The new change is passing pytest and the production run checks I have.